src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplate.java

```diff
@@ -575,11 +573,17 @@ private DockerTransientNode doProvisionNode(DockerAPI api, DockerClient client,
             node.setDockerAPI(api);
             finallyRemoveTheContainer = false;
             return node;
-        } finally {
+        }
+        catch(Throwable e) {
```

Comment:
I don't like the idea of catching `Throwable`.
I'd suggest you just catch whatever checked exceptions that code can throw, plus `RuntimeException`.